### DISCUSSION TITLE
JSON metadata description

### DISCUSSION THREAD

**Discussion by vpinna80** [VTL-Community/VTL-Community#39] [Ideas]
This discussion is to gather all the information on possible formats for the JSON metadata description file.
This JSON file is intended as a placeholder to have a harmonized description of the VTL Information Model inside the various tools until a decision on this is finalized by the VTL TF.

**Comment by vpinna80** [comment]
[example_valuedomain.json](https://github.com/VTL-Community/VTL-Community/files/13264753/example_valuedomain.json)
[example_datastructure.json](https://github.com/VTL-Community/VTL-Community/files/13264755/example_datastructure.json)
[metadata1.json](https://github.com/VTL-Community/VTL-Community/files/13264760/metadata.json)
[metadata2.json](https://github.com/VTL-Community/VTL-Community/files/13264765/metadata.json)

**Reply by javihern98** [reply]
As a first approach to the discussion:

- Mapping between data files and data structure, do we need it inside the JSON?
- Single JSON for all metadata (datastructure and valuedomains), this can be a nice approach but less modular
- Definition of Constraints on VTL, we did not consider this as it may be a validation using the VTL Language.

**Comment by vpinna80** [comment]
The first I can't answer, I think, where would it be placed?
The second I think we can also have it both ways or in separate files, I don't think it really matters in the end
The third comes from a use case that we had in BoI writing a study, the VTL has this concept of "described subsets" but it's mostly unused. It could be used to implicitly validate results, in the IN expression, in casts, to translate the SQL LIKE, and so on.
